### PR TITLE
doc: release: 3.7: Add mcuboot/build system changes

### PR DESCRIPTION
**Pull request by nordicjm**

### CODE REVIEW

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -255,6 +255,15 @@ Build system and Infrastructure
 
   * Fixed issue with passing UTF-8 configs to applications using sysbuild.
 
+  * Fixed issue whereby domain file in sysbuild projects would be loaded and used with outdated
+    information if sysbuild configuration was changed then ``west flash`` was ran directly after.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:259]: would it be a bit better like so:
```suggestion
    information if sysbuild configuration was changed, and ``west flash`` was ran directly after.
```
?

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -255,6 +255,15 @@ Build system and Infrastructure
...
+
+  * Fixed issue with zephyr modules not being listed in sysbuild if they did not have a Kconfig
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:261]:
```suggestion
  * Fixed issue with Zephyr modules not being listed in sysbuild if they did not have a Kconfig
```

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -659,6 +668,42 @@ HALs
...
+
+  * Fixed issue with swap-move algorithm failing to validate multiple-images
+
+  * Fixed align script error in imgtool
+
+  * Fixed img verify for hex file format in imgtool
+
+  * Fixed issue with reading the flash image reset vector
+
+  * Fixed too-early check_config.h include in mbedtls
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:687]:
```suggestion
  * Fixed too-early ``check_config.h`` include in mbedtls
```
?

**Review by aescolar — DISMISSED** [review]
Thanks, just nits

**Review by aescolar — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**